tasks/eth/008-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,263 @@
+        _proxies.OptimismMintableERC20Factory =
+            stdJson.readAddress(addressesJson, "$.OptimismMintableERC20FactoryProxy");
+        _proxies.OptimismPortal = stdJson.readAddress(addressesJson, "$.OptimismPortalProxy");
+        _proxies.OptimismPortal2 = stdJson.readAddress(addressesJson, "$.OptimismPortalProxy");
+        _proxies.SystemConfig = stdJson.readAddress(addressesJson, "$.SystemConfigProxy");
+        _proxies.L1ERC721Bridge = stdJson.readAddress(addressesJson, "$.L1ERC721BridgeProxy");
+        _proxies.DisputeGameFactory = dgfProxy;
+
+        // Read superchain.yaml
+        string[] memory inputs = new string[](4);
+        inputs[0] = "yq";
+        inputs[1] = "-o";
+        inputs[2] = "json";
+        inputs[3] = "lib/superchain-registry/superchain/configs/mainnet/superchain.yaml";
+
+        addressesJson = string(vm.ffi(inputs));
+
+        _proxies.ProtocolVersions = stdJson.readAddress(addressesJson, "$.protocol_versions_addr");
+        _proxies.SuperchainConfig = stdJson.readAddress(addressesJson, "$.superchain_config_addr");
+    }
+ 
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory simPayload)
+        internal
+        view
+        override
+    {
+        console.log("Running post-deploy assertions");
+        checkSemvers();
+        checkStateDiff(accesses);
+        checkSystemConfig();
+        checkOptimismPortal();
+        console.log("All assertions passed!");
+    }
+
+    function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view override {
+        require(accountAccesses.length > 0, "No account accesses");
+
+        super.checkStateDiff(accountAccesses);
+
+        // Assert that no other OP Contracts other than the SystemConfig and the OptimismPortal are accessed
+        for (uint256 i; i < accountAccesses.length; i++) {
+            Vm.AccountAccess memory accountAccess = accountAccesses[i];
+            require(
+                accountAccess.oldBalance == accountAccess.newBalance,
+                string.concat("Unexpected balance change: ", vm.toString(accountAccess.account))
+            );
+            require(
+                accountAccess.kind != VmSafe.AccountAccessKind.SelfDestruct,
+                string.concat("Self-destructed account: ", vm.toString(accountAccess.account))
+            );
+
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                Vm.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+
+                address account = storageAccess.account;
+                require(account != proxies.L1CrossDomainMessenger, "state-000");
+                require(account != proxies.L1ERC721Bridge, "state-100");
+                require(account != proxies.L1StandardBridge, "state-200");
+                require(account != proxies.OptimismMintableERC20Factory, "state-300");
```

Comment:
The comment says: "Assert that no other OP Contracts other than the SystemConfig and the OptimismPortal are accessed".

This check could be strengthened by requiring that the `account` is one of a number of values, rather than that it is not some value.

Reply:
> or I guess read the safe addresses from .env

👍